File: client-app/src/app/models/Game.ts
import { Move } from "./Move";
export interface Game {
  id?: number;
  player1: string;
  player2: string;
  currentTurn: string; // "Player1" | "Player2" | "Computer"
  gameType: GameType;
  status: string;      // "Active" | "Completed" | "Reset"
  moves?: Move[];
}

export type GameType = 'PvP' | 'PvC'

File: TicTacToeApp/API/Models/Game.cs
namespace API.Models
{
    public class Game
    {
        public int Id { get; set; }

        public Guid SessionId { get; set; }
        public string Player1 { get; set; }
        public string Player2 { get; set; }

        public string GameType { get; set; } // "PvP" or "PvC"
        public string CurrentTurn { get; set; } // Player1 / Player2 / Computer
        public string Status { get; set; } // Active, Completed, Reset
        public ICollection<Move> Moves { get; set; }
    }
}